src/lib.rs

```diff
@@ -467,8 +461,8 @@ impl<T: Config> Pallet<T> {
 		let mut sum_outcome = BalanceOf::<T>::default();
 		let mut sum_deposit = BalanceOf::<T>::default();
 
-		for (i, part) in parts.iter().enumerate() {
-			let fid = Self::calc_funding_id(channel, part);
+		for (i, part) in parts.iter().cloned().enumerate() {
+			let fid = FundingOf::<T> { channel, part }.id::<HasherOf<T>>();
```

Comment:
I think I like using the explicit method better than creating a funding object here. I suggest you keep the method and do the respective operation there.

Reply:
Yes i could also do that, it probably reads easier.  
I will still keep the `Funding` (maybe rename) type since i need it in the connector.